EIPS/eip-6077.md

```diff
@@ -0,0 +1,75 @@
+---
+eip: 6077
+title: Invalidation for Signature-Based Operations
+description: Extends EIP-712 and unifies EIP-2612 with many others
+author: Anton Bukov (@k06a), Mikhail Melnik (@zumzoom)
+discussions-to: https://ethereum-magicians.org/t/eip-6077-invalidation-abstraction-for-signature-based-operations/12162
+status: Draft
+type: Standards Track
+category: ERC
+created: 2022-12-02
+requires: 712, 2612
+---
+
+## Abstract
+
+This EIP extends [EIP-712](./eip-712.md) and defines two methods to validate and invalidate signature-based operations, such as [EIP-2612](./eip-20.md)'s permit operation. This EIP provides two ways to track such operations and invalidate them:
+
+- main sequnce of incremental nonces per signer
+- sequence of incremental ids per signer per beneficiary
+
+## Motivation
+
+Same abstraction could be utilized by mutilple signature-based operations, moreover existing EIPs like [EIP-2612](./eip-2612.md) could be considered as fully compatible with the EIP.
+
+Multiple EIPs define operations that can be executed in behalf of signer and sometime introduce method naming collisions and other. For example, [EIP-2612](./eip-2612.md) defines both `permit` and `nonces` methods, but gives no clue that `nonces` is related to permit operation. In case of multiple same-level EIPs implemented within one smart contract (for example: permit, delegate, vote) it's obvious that they should use different nonces.
```

Comment:
I can't understand what you're trying to say here. Mind explaining?

Reply:
Trying to create abstraction for signature-based operations, and this abstraction also covers existing EIP-2612 somehow.